### ISSUE TITLE
[a11y]: ContentSwitcher [icon only with layer & with layer] - tab set size announced incorrectly

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14051]
### Package

@carbon/react

### Browser

Safari & iPhone

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO

### Description

VO in iPhone Safari announces the posinset and setsize wrongly

### WCAG 2.1 Violation

1.3.1 Info & Relationships

### Reproduction/example

Record `Number 52` in the below table: 
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

When the tab control receives SR focus in Left/Right swipe, SR announces the total number of tabs and the index possition of the current tab wrongly. e.g. when each tab receives focus, SR announces "tab 1 of 1" instead of announcing "tab 1 of 3", "tab 2 of 3" etc.

### User Impact
When the current index position of the tab and the total number of tabs announced wrongly, the user may get confused

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
